subliminal/providers/subscenter.py

```diff
@@ -0,0 +1,180 @@
+            raise ConfigurationError('Username and password must be specified')
+        self.username = username
+        self.password = password
+        self.session = None
+        self.logged_in = False
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers = {'User-Agent': 'Subliminal/%s' % __version__.split('-')[0]}
+        # login
+        if self.username is not None and self.password is not None:
+            logger.debug('Logging in')
+            url = self.server + 'subscenter/accounts/login/'
+            # Retrieve CSRF token first.
+            self.session.get(url)
+            csrf_token = self.session.cookies['csrftoken']
+            data = {'username': self.username, 'password': self.password,
+                    'next': '/he/', 'csrfmiddlewaretoken': csrf_token}
+            r = self.session.post(url, data, timeout=10, allow_redirects=False)
+            if r.status_code == 302:
+                logger.info('Logged in')
+                self.logged_in = True
+            else:
+                raise AuthenticationError(self.username)
+
+    def terminate(self):
+        # logout
+        if self.logged_in:
+            r = self.session.get(self.server + 'subscenter/accounts/logout/', timeout=10)
+            logger.info('Logged out')
+            if r.status_code != 200:
+                raise ProviderError('Request failed with status code %d' % r.status_code)
+            self.logged_in = False
+        self.session.close()
+
+    @staticmethod
+    def slugify(string):
+        new_string = string.replace(' ', '-').replace("'", '').replace(':', '').lower()
+        # We remove multiple spaces by using this regular expression.
+        return re.sub('-+', '-', new_string)
+
+    def query(self, languages=None, series=None, season=None, episode=None, title=None):
+        # Converts the title to Subscenter format by replacing whitespaces and removing specific chars.
+        if series and season and episode:
+            # Search for a TV show.
+            kind = 'episode'
+            slugified_series = self.slugify(series)
+            url = self.server + 'cinemast/data/series/sb/' + slugified_series + '/' + str(season) + '/' + \
+                str(episode) + '/'
+            page_link = self.server + 'subtitle/series/' + slugified_series + '/' + str(season) + '/' + \
+                str(episode) + '/'
+        elif title:
+            # Search for a movie.
+            kind = 'movie'
+            slugified_title = self.slugify(title)
+            url = self.server + 'cinemast/data/movie/sb/' + slugified_title + '/'
+            page_link = self.server + 'subtitle/movie/' + slugified_title + '/'
+        else:
+            raise ValueError('One or more parameters are missing')
+        logger.debug('Searching subtitles %r', {'title': title, 'season': season, 'episode': episode})
+        response = self.session.get(url)
```

Comment:
Ohhh ok I think I understand what you mean.
The problem with this approach is that the site makes you log in before presenting the options, and since I prefer the login part to always be optional, I think this is the best way.